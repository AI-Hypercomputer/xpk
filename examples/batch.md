### Running batch tasks in xpk

### Preparation

This section should describe how to prepare volumes and add external files executed in script.

### Execution

To run xpk batch script run:
```bash
python3 xpk.py batch --cluster my-cluster slurm_sample.sh
```

To view results run:
```bash
kubectl get pods
```
Names of pods from slurm task execution begin with "xpk-def-app-profile-slurm".To see pods on which batch task was executed.

To see logs run:
```bash
kubectl logs <pod_name>
```

