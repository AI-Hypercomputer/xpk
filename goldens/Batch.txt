$ python3 xpk.py batch --project=golden-project --zone=us-central1-a --cluster=golden-cluster --dry-run batch-read.sh
[XPK] Starting xpk
[XPK] Working on golden-project and us-central1-a
[XPK] Try 1: get-credentials to cluster golden-cluster
[XPK] Task: `get-credentials to cluster golden-cluster` is implemented by the following command not running since it is a dry run. 
gcloud container clusters get-credentials golden-cluster --region=us-central1 --project=golden-project && kubectl config view && kubectl config set-context --current --namespace=default
[XPK] Task: `GKE Cluster Get ConfigMap` is implemented by the following command not running since it is a dry run. 
kubectl get configmap golden-cluster-resources-configmap -o=custom-columns="ConfigData:data" --no-headers=true
[XPK] Task: `GKE Cluster Get ConfigMap` is implemented by the following command not running since it is a dry run. 
kubectl get configmap golden-cluster-resources-configmap -o=custom-columns="ConfigData:data" --no-headers=true
[XPK] Task: `GKE Cluster Get ConfigMap` is implemented by the following command not running since it is a dry run. 
kubectl get configmap golden-cluster-metadata-configmap -o=custom-columns="ConfigData:data" --no-headers=true
[XPK] Task: `submit job` is implemented by the following command not running since it is a dry run. 
kubectl kjob create slurm --profile xpk-def-app-profile --localqueue multislice-queue --worker-container xpk-batch-container --first-node-ip --pod-template-annotation kueue.x-k8s.io/podset-preferred-topology=cloud.google.com/gce-topology-host -- batch-read.sh --partition multislice-queue
[XPK] XPK Done.
